File: 0733-flood-fill/0733-flood-fill.java
class Solution {
    public void dfs(int[][] image, int sr, int sc, int color, int keepColor){
        if(sr < 0 || sr >= image.length || sc < 0 || sc >= image[0].length || image[sr][sc] != keepColor) return; 

        image[sr][sc] = color; 
        dfs(image,sr+1,sc,color,keepColor); 
        dfs(image,sr-1,sc,color,keepColor);
        dfs(image,sr,sc+1,color,keepColor); 
        dfs(image,sr,sc-1,color,keepColor); 
    }
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        if (image[sr][sc] == color) return image;
        dfs(image,sr,sc,color,image[sr][sc]);
        return image; 
    }
}

File: 0733-flood-fill/0733-flood-fill.cpp
class Solution {
public:

    void dfs(vector<vector<int>>& image, int sr, int sc, int color, int startnum){
        if (sr >= image.size() || sr < 0){
            return;
        }
        if (sc >= image[sr].size() || sc < 0){
            return;
        }
        
        if (image[sr][sc] != startnum){
            return;
        }
        
        image[sr][sc] = color;
        dfs(image,sr+1,sc,color,startnum);
        dfs(image,sr-1,sc,color,startnum);
        dfs(image,sr,sc+1,color,startnum);
        dfs(image,sr,sc-1,color,startnum);
    }

    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        if (image[sr][sc] == color){
            return image;
        }
        int startnum = image[sr][sc];
        dfs(image,sr,sc,color,startnum);
        return image;
        
    }
};